### PR TITLE
Figure out host for span from cluster instead of weighted cluster

### PR DESCRIPTION
**Pull request by bianpengyuan**
Misused weighted cluster for host name in trace span.

### CODE REVIEW

File: pilot/pkg/networking/core/v1alpha3/route/route.go [hunk 1]
```diff
@@ -502,14 +502,13 @@ func getRouteOperation(in *route.Route, vsName string, port int) string {
...
 
-	operation := fmt.Sprintf("%s:%d%s", vsName, port, path)
 	// use host:port + match uri if there is only one destination cluster.
-	wc := in.GetRoute().GetWeightedClusters()
-	if c := wc.GetClusters(); len(c) == 1 {
-		_, _, h, p := model.ParseSubsetKey(c[0].Name)
-		operation = fmt.Sprintf("%s:%d%s", h, p, path)
+	if c := in.GetRoute().GetCluster(); c != "" {
+		_, _, h, p := model.ParseSubsetKey(c)
+		return fmt.Sprintf("%s:%d%s", h, p, path)
```
Comment (rshriram) [inline pilot/pkg/networking/core/v1alpha3/route/route.go]: ? Please add comments. This entire file is poorly documented, with unintuitive function names that its hard to understand whats going on.

Comment (kyessenov) [reply pilot/pkg/networking/core/v1alpha3/route/route.go]: GetRouteOperation derives a human-readable descriptor for an envoy route...

**Comment by kyessenov** [comment]
Please add a comment on the function. I think that should satisfy @rshriram 's request.

File: pilot/pkg/networking/core/v1alpha3/route/route.go (hunk 1 above)
Comment (rshriram) [reply pilot/pkg/networking/core/v1alpha3/route/route.go]: That much is evident. What is the new change doing?

**Comment by bianpengyuan** [comment]
@rshriram ptal?

File: pilot/pkg/networking/core/v1alpha3/route/route.go [hunk 2]
```diff
@@ -535,14 +535,16 @@ func getRouteOperation(in *route.Route, vsName string, port int) string {
...
-	// use host:port + match uri if there is only one destination cluster.
-	wc := in.GetRoute().GetWeightedClusters()
-	if c := wc.GetClusters(); len(c) == 1 {
-		_, _, h, p := model.ParseSubsetKey(c[0].Name)
-		operation = fmt.Sprintf("%s:%d%s", h, p, path)
-	}
-	return operation
+	// If there is only one destination cluster in route, return host:port/uri as description of route.
+	// Otherwise there are multiple destination clusters and destination host is not clear. For that case
+	// return virtual serivce name:port/uri as substitute.
```
Comment (rshriram) [inline pilot/pkg/networking/core/v1alpha3/route/route.go]: Nicely written thanks.

**Pull request merged**